src/ast_pattern.ml

```diff
@@ -220,5 +220,15 @@ let elist (T f) = T (fun ctx _loc e k ->
   k (List.map l ~f:(fun x -> f ctx x.Parsetree.pexp_loc x (fun x -> x))))
 ;;
 
+let esequence (T f) = T (fun ctx _loc e k ->
+  let rec parse_seq expr =
+    match expr.pexp_desc with
+    | Pexp_sequence (expr, next) ->
+      expr :: (parse_seq next)
+    | _ -> [ expr ]
```

Comment:
Do you think it might be worth making a tail recursion out if this?

Reply:
Similarly to [here](https://github.com/ocaml-ppx/ppxlib/blob/master/src/ast_pattern.ml#L209).